src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Analyzer.java

```diff
@@ -783,5 +789,19 @@ public JCTree visitNewClass(NewClassTree node, Void aVoid) {
             }
             return newTree;
         }
+
+        @Override
+        public JCTree visitVariable(VariableTree node, Void p) {
+            JCTree result = super.visitVariable(node, p);
+            if (node == wrappedTree) {
+                //The current tree is a field and has been wrapped by a block, so it effectivelly
+                //became local variable. If it has some modifiers (except for final), an error
+                //would be reported, causing the whole rewrite to fail. Removing the non-final
+                //modifiers from the variable here:
+                ((JCVariableDecl) result).mods.flags &= Flags.FINAL;
```

Comment:
just curious: why preserving the FINAL flag anyway?

Reply:
No particular strong reason, just keeping what we can keep. We can clear the flags to `0`, if you prefer. I don't think that should cause issues (with the current set of Analyzers).

Reply:
I'm fine with either version was just curious, do as you consider best